executor/runtime/docker/docker.go

```diff
@@ -1955,32 +1988,32 @@ func (r *DockerRuntime) Kill(c *runtimeTypes.Container) error { // nolint: gocyc
 		goto stopped
 	}
 
-	if err := r.client.ContainerStop(context.TODO(), c.ID, &containerStopTimeout); err != nil {
+	if err := r.client.ContainerStop(context.TODO(), r.c.ID, &containerStopTimeout); err != nil {
 		r.metrics.Counter("titus.executor.dockerStopContainerError", 1, nil)
-		log.Errorf("container %s : stop %v", c.TaskID, err)
+		log.Errorf("container %s : stop %v", r.c.TaskID, err)
 		errs = multierror.Append(errs, err)
 	} else {
 		goto stopped
 	}
 
-	if err := r.client.ContainerKill(context.TODO(), c.ID, "SIGKILL"); err != nil {
+	if err := r.client.ContainerKill(context.TODO(), r.c.ID, "SIGKILL"); err != nil {
```

Comment:
Same as above

Reply:
No, we can't, because by the time that this part (Kill) gets called, the parent context may already have been cancelled (hence why we're killing. I don't yet know how to solve this problem. Suggestions welcome.